server/src/main/java/org/elasticsearch/search/aggregations/metrics/PercentilesAggregationBuilder.java

```diff
@@ -21,154 +21,86 @@
-    public static final ParseField NUMBER_SIGNIFICANT_DIGITS_FIELD = new ParseField("number_of_significant_value_digits");
+    private static final ParseField PERCENTS_FIELD = new ParseField("percents");
 
-    private static class TDigestOptions {
-        Double compression;
-    }
-
-    private static final ObjectParser<TDigestOptions, Void> TDIGEST_OPTIONS_PARSER =
-            new ObjectParser<>(PercentilesMethod.TDIGEST.getParseField().getPreferredName(), TDigestOptions::new);
+    private static final ConstructingObjectParser<PercentilesAggregationBuilder, String> PARSER;
     static {
-        TDIGEST_OPTIONS_PARSER.declareDouble((opts, compression) -> opts.compression = compression, COMPRESSION_FIELD);
-    }
+        PARSER = new ConstructingObjectParser<>(PercentilesAggregationBuilder.NAME, false, (objects, name) -> {
 
-    private static class HDROptions {
-        Integer numberOfSigDigits;
-    }
+            double[] values = ((List<Double>) objects[0]).stream().mapToDouble(Double::doubleValue).toArray();
+            PercentilesMethod.Config percentilesConfig;
 
-    private static final ObjectParser<HDROptions, Void> HDR_OPTIONS_PARSER =
-            new ObjectParser<>(PercentilesMethod.HDR.getParseField().getPreferredName(), HDROptions::new);
-    static {
-        HDR_OPTIONS_PARSER.declareInt(
-                (opts, numberOfSigDigits) -> opts.numberOfSigDigits = numberOfSigDigits,
-                NUMBER_SIGNIFICANT_DIGITS_FIELD);
-    }
+            if (objects[1] != null && objects[2] != null) {
+                throw new IllegalStateException("Only one percentiles method should be declared.");
+            } else if (objects[1] == null && objects[2] == null) {
+                // Default is tdigest
+                percentilesConfig = new PercentilesMethod.Config.TDigest();
+            } else if (objects[1] != null) {
+                percentilesConfig = (PercentilesMethod.Config) objects[1];
+            } else {
+                percentilesConfig = (PercentilesMethod.Config) objects[2];
+            }
 
-    private static final ObjectParser<InternalBuilder, Void> PARSER;
-    static {
-        PARSER = new ObjectParser<>(PercentilesAggregationBuilder.NAME);
-        ValuesSourceParserHelper.declareAnyFields(PARSER, true, true);
+            return new PercentilesAggregationBuilder(name, values, percentilesConfig);
+        });
 
-        PARSER.declareDoubleArray(
-                (b, v) -> b.percentiles(v.stream().mapToDouble(Double::doubleValue).toArray()),
-                PERCENTS_FIELD);
+        ValuesSourceParserHelper.declareAnyFields(PARSER, true, true);
 
+        PARSER.declareDoubleArray(ConstructingObjectParser.constructorArg(), PERCENTS_FIELD);
         PARSER.declareBoolean(PercentilesAggregationBuilder::keyed, KEYED_FIELD);
+        PARSER.declareObject(ConstructingObjectParser.optionalConstructorArg(), PercentilesMethod.TDIGEST_PARSER,
```

Comment:
I think we should have *something* in the superclass that helps with the parsing. I'm not sure what though.